### ISSUE TITLE
Select Field values not duplicated without prior save

### ISSUE THREAD

**Issue by shazzygh** [mcguffin/acf-duplicate-repeater#3]
When duplicating a row containing a select field, it will only duplicate the select field if the post (and content of the field) has been saved before. 

Steps to reproduce:
1. Have repeater field containing Select Field
2. Change Select Field without saving the post
3. Duplicate Row

*edit*: This is likely the cause of this issue:
> Note: For performance reasons, the dynamic state of certain form elements (e.g., user data typed into textarea and user selections made to a select) is not copied to the cloned elements. When cloning input elements, the dynamic state of the element (e.g., user data typed into text inputs and user selections made to a checkbox) is retained in the cloned elements.

https://api.jquery.com/clone/

**Comment by shazzygh** [comment]
I was able to come up with solutions for successfully duplicating select, wysiwyg editor and datepicker fields (see CUSTOM additions in the modified code below). Again, the coding can likely be improved, but it gets the job done.

```javascript
_duplicate: function( e ) {

var $prev_clone, 
	id, uniquid,
	$row, 
	$template, $tmp,
	$copy;
// vars


// add before row
if( e.$el.hasClass('acf-icon') ) {
	$row = e.$el.closest('.acf-row');
}

// backup acf clone options
$prev_clone = this.$clone;

$template = $row.clone();
$tmp = $('<table />').appendTo('body').append( $template );

reset_ids( $template );

reset_fields( $template, $row.closest('.acf-table').find('tr.acf-clone') );

// fake acf.clone() current row
this.$clone = {
	'$el'		: $template,
	'search'	: 'acfcloneindex',
	'replace'	: false,
};

// CUSTOM: get selection values
var selectedValue = {};
$row.find('select').each(function(index) {
	selectedValue[index] = $(this).val();    
});

// CUSTOM: get WYSIWYG editor contents
var $editorContent = {};
$row.find('.acf-field-wysiwyg iframe').contents().find("body").each(function(index) {
	$editorContent[index] = $(this).html();
});

// make copy
$copy = this.add( $row );

// restore acf clone options
this.$clone = $prev_clone;
$tmp.remove();

// CUSTOM: restore selection values
$copy.find('select').each(function(index) {
	$(this).find("option[value = '" + selectedValue[index] + "']").attr("selected", "selected");
});

// CUSTOM: restore WYSIWYG editor contents
$copy.find('.acf-field-wysiwyg textarea').each(function(index) {
	$(this).html($editorContent[index]);
});

// CUSTOM: reinit Datepicker, customize args to your liking
$copy.find('.hasDatepicker').each(function() {

	var $datepickerParent = $(this).parent();
	var setDateFormat = $datepickerParent.data('date_format');
	var setFirstDay = $datepickerParent.data('first_day');
	var setAltField = $datepickerParent.find('input[type="hidden"]');

	var args = { 
		dateFormat:			setDateFormat,
		altField:			setAltField,
		altFormat:			'yymmdd',
		changeYear:			true,
		yearRange:			"-100:+100",
		changeMonth:			true,
		showButtonPanel:		true,
		firstDay:			setFirstDay
	};

	$(this).removeAttr('id').removeClass('hasDatepicker').datepicker(args);

});



// init fields
//*/
// give the copy back
return $copy;

}
```

**Comment by mcguffin** [comment]
Hi and thanks for looking into it!
Would you mind to put this in a pull request...?

**Comment by mcguffin** [comment]
Thanks for the PR! It works as far as I can see. It's included in release 0.1.5.
